Code/GraphMol/SmilesParse/SmartsWrite.cpp

```diff
@@ -656,7 +656,26 @@ std::string FragmentSmartsConstruct(ROMol &mol, unsigned int atomIdx,
   for (auto &atom : mol.atoms()) {
     atom->updatePropertyCache(false);
   }
-  Canon::canonicalizeFragment(mol, atomIdx, colors, ranks, molStack);
+
+  // Another dirty trick to avoid reordering of lead chiral atoms in
+  // canonicalizeFragment: since we are writing SMARTS, there won't be a
+  // reordering on parsing.
+  // The trick consists in just avoiding the chiral atom to be the first in the
+  // SMARTS string by swapping atomIdx to the first bonded atom. This will cause
+  // the neighbor to go first.
+  if (atomIdx == 0) {
+    const Atom *atom = mol.getAtomWithIdx(atomIdx);
+    if (atom->getChiralTag() == Atom::CHI_TETRAHEDRAL_CCW ||
+        atom->getChiralTag() == Atom::CHI_TETRAHEDRAL_CW) {
+      auto nbrItr = mol.getAtomNeighbors(atom);
+      if (nbrItr.first != nbrItr.second) {
+        atomIdx = mol[*nbrItr.first]->getIdx();
```

Comment:
This is a clever, lightly hacky, solution to the problem.
There's a very minor chance that this could still fail if the neighbor you pick also happens to be chiral. I'm not sure whether or not it's worth worrying about.

Reply:
Oh, you are right. I completely missed that the neighbors might also be chiral. I think we should consider that. I'll turn that into a loop to find the first non-chiral atom and start the SMARTS from there.